### PR TITLE
Some ESAPI Updates

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
Mostly very small commits.
Some error fixing in session handling and esys context.

### CODE REVIEW

File: src/tss2-esys/esys_iutil.c [hunk 1]
```diff
@@ -427,6 +427,14 @@ iesys_get_handle_type(TPM2_HANDLE handle)
     return ht;
 }
 
+/** Compute name derived from public info with tpm name.
```
Comment (danintel) [inline src/tss2-esys/esys_iutil.c]: s/Compute/Compare/
s/with tpm name./with a tpm name./

File: src/tss2-esys/esys_iutil.h [hunk 2]
```diff
@@ -71,6 +71,13 @@ extern "C" {
...
     }
 
+#define goto_if_null(p,msg,ec,label) \
+    if (p == NULL) { \
+        LOG_ERROR("%s ", msg); \
+        r = ec; \
+        goto label;  \
+    }
+
 #define goto_if_error(r,msg,label) \
```
[the remark this answers is not in the document]
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.h]: done

File: src/tss2-esys/esys_iutil.c (hunk 1 above)
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.c]: done

File: src/tss2-esys/esys_iutil.c [hunk 3]
```diff
@@ -896,12 +896,20 @@ iesys_is_object_bound(const TPM2B_NAME * name,
...
+ * and the auth value. The auth value is appended to the session key.
+ * The session value is used for key derivation for parameter encryption and
+ * HMAC computation. There is one exception for HMAC key derivation: If the
+ * session is bound to an object only the session key is used. The auth value
+ * is appended only for the key used for parameter encryption.
  * The auth value is only used if an authorization is necessary and the name
  * of the object is not equal to the name of an used bound entity
- * @param[in] session for which the session value will be computed
+ * @param[in,out] session for which the session value will be computed.
+ *       The value will be stored in stored in sessionValue of the session object.
```
[the remark this answers is not in the document]
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_iutil.c]: done

File: src/tss2-esys/esys_mu.c [hunk 4]
```diff
@@ -576,6 +576,11 @@ Tss2_MU_IESYS_SESSION_Marshal(
         LOG_ERROR("Error marshaling subfield sessionValue");
         return ret;
     }
+    ret = Tss2_MU_UINT16_Marshal(src->sizeHmacValue, buffer, size, &offset_loc);
+    if (ret != TSS2_RC_SUCCESS) {
```
[the remark this answers is not in the document]
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_mu.c]: done
... also for the rest of the file

File: src/tss2-esys/esys_mu.c [hunk 5]
```diff
@@ -707,6 +712,13 @@ Tss2_MU_IESYS_SESSION_Unmarshal(
         LOG_ERROR("Error unmarshaling subfield sessionValue");
         return ret;
     }
+    UINT16 out_sizeHmacValue;
+    ret = Tss2_MU_UINT16_Unmarshal(buffer, size, &offset_loc,
+            (dst == NULL)? &out_sizeHmacValue : &dst->sizeHmacValue);
+    if (ret != TSS2_RC_SUCCESS) {
```
[the remark this answers is not in the document]
Comment (AndreasFuchsSIT) [reply src/tss2-esys/esys_mu.c]: done
... also for the rest of the file

**Comment by AndreasFuchsSIT** [comment]
All comments addressed.
I did put 2 trivial additional commits into this PR:
- Extending the return_if_error() scheme from the comment to the whole file
- Rename the functions of esys_mc

**Pull request merged**